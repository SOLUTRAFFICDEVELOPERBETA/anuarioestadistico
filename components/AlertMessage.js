import React from 'react';
import { IconButton, makeStyles, Snackbar } from '@material-ui/core';
import { Close } from '@material-ui/icons';
import { green, red, deepOrange } from '@material-ui/core/colors';
import AlertContext from '../contexts/alert';

// Estilos del componente
const useStyles = makeStyles((theme) => ({
    root: {
        position: 'fixed',
        top: 0,
        left: 0,
        padding: theme.spacing(1.5),
        width: '100%',
        zIndex: 1600,

        [theme.breakpoints.down('md')]: {
            width: 'auto',
            height: 'fit-content',
            bottom: theme.spacing(2),
            right: theme.spacing(3),
            borderRadius: theme.spacing(1),
            left: 'auto',
            margin: theme.spacing(1, 0)
        }
    },
    message: {
        display: 'grid',
        alignItems: 'center',
        gridGap: theme.spacing(1)
    },
    success: {
        backgroundColor: green[500],
        color: theme.palette.getContrastText(green[500])
    },
    error: {
        backgroundColor: red[500],
        color: theme.palette.getContrastText(red[500])
    },
    warning: {
        backgroundColor: deepOrange[500],
        color: theme.palette.getContrastText(deepOrange[500])
    }
}));

/**
 * @description Componente que se encarga de mostrar el mensaje de cualquier notificaciÃ³n de la plataforma.
 * @see AlertContext
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const AlertMessage = () => {
    const classes = useStyles();
    const { closeMessage } = React.useContext(AlertContext);

    return (
        <AlertContext.Consumer>
            {({ message }) =>
                message ? (
                    <Snackbar
                        open={Boolean(message)}
                        autoHideDuration={message.time}
                        onClose={closeMessage}
                        anchorOrigin={{
                            vertical: 'bottom',
                            horizontal: 'center'
                        }}
                        ContentProps={{
                            classes: {
                                root: classes[message.category],
                                message: classes.message
                            }
                        }}
                        message={message.text}
                        action={
                            <React.Fragment>
                                <IconButton size="small" color="inherit" onClick={closeMessage}>
                                    <Close fontSize="small" />
                                </IconButton>
                            </React.Fragment>
                        }
                    />
                ) : null
            }
        </AlertContext.Consumer>
    );
};

export default AlertMessage;
