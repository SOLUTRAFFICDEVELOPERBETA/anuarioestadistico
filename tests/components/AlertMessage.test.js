import { mount } from "enzyme"
import '@testing-library/jest-dom'
import AlertMessage from "../../components/AlertMessage"
import AlertContext from "../../contexts/alert"

describe('Pruebas en <AlertMessage/>', () => {
    const closeMessage = jest.fn();
    const message = {
        text: 'Texto de prueba',
        category: 'succes',
        time: 3000
    }
    const wrapper = mount(
        <AlertContext.Provider value={{
            closeMessage,
            message

        }}>

            <AlertMessage
            />

        </AlertContext.Provider>
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostar el mensaje correctamente', () => {
        expect(wrapper.find('div').at(2).text().trim()).toBe(message.text);
    });

    test('debe llamar a la funciÃ³n closeMessage()', () => {
        wrapper.find('button').prop('onClick')()
        expect(closeMessage).toHaveBeenCalled();
        expect(closeMessage).toHaveBeenCalledTimes(1);
    })
})