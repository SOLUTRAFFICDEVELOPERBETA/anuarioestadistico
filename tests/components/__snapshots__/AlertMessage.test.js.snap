// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AlertMessage/> debe mostrarse correctamente 1`] = `
<div
  className="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
>
  <div
    className="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
    direction="up"
    role="alert"
    style={
      Object {
        "opacity": 1,
        "transform": "scale(1, 1)",
        "visibility": undefined,
      }
    }
  >
    <div
      className="MuiSnackbarContent-message makeStyles-message-2"
    >
      Texto de prueba
    </div>
    <div
      className="MuiSnackbarContent-action"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
